### PR TITLE
[2.19.x] DDF-UI-309 Result form collection creates empty items

### PR DESCRIPTION
**Pull request by abel-connexta**
ddf-ui PRs:
3.4.x codice/ddf-ui#310

---------------------
#### What does this PR do?
Issue 1 - Result form collection creates empty items when it shouldn't for Guest
Issue 2 - Not able to set default a Result form item (already fixed in 2.19.x)

#### Who is reviewing it? 

@andrewzimmer 
@hayleynorton
@cassandrabailey293 
@zta6 

#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining
@mojogitoverhere
#### How should this be tested?

Issue 1:
You should be logged in as Guest (Confirm by checking on upper right corner)
1. Go to Result Forms
2. Click on "New Result Forms" to add new result form
3. Ensure you get two error pop up messages, and ensure no Result From Box are created

Issue 2:
1. Log in as admin (Confirm by checking on upper right corner)
2. Ensure in Search form no default is selected for existing Search forms, or make sure no Search forms are created
3. Go to Result form
4. Click on "New Result Forms" to add new result form
5. Make created result form as default
6. Ensure no error on console and the star icon appears and the 3 dots setting on lower right corner does not disappear.
7. Clear default Result form
8. Ensure no regression
9. Do step 3 to 8 for Search Form

#### What are the relevant tickets?
codice/ddf-ui#309

##### Before
Before-Result-Form-Guest
Before-Result-Admin-Default

##### After
After-Result-Form-Guest

After-Result-Admin-Default

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/component/result-form/result-form.view.js [hunk 1]
```diff
@@ -179,8 +179,10 @@ module.exports = Marionette.LayoutView.extend({
       error: () => {
         this.errorMessage()
       },
+      wait: true,
```
Comment (cassandrabailey293) [inline ui/packages/catalog-ui-search/src/main/webapp/component/result-form/result-form.view.js:182]: ❓ why did you need the wait?

Comment (abel-connexta) [reply ui/packages/catalog-ui-search/src/main/webapp/component/result-form/result-form.view.js:182]: This is waiting for the sent out request to backend to respond before adding the Result Form. Without the wait, it immediately adds the Result Form box without the content inside, and if it returns with an error, you are just left with that empty box.

**Comment by bdeining** [comment]
build now

**Comment by zta6** [comment]
🎉 Hero Successful 🎉 
Verified 'default' functionality works for both search forms and result forms
Unable to test issue 1 because of inability to log in as Guest on 2.19.x
